### ISSUE TITLE
stdout is broken after `pylint.lint.Run`

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#5702]
### Bug description

This is `bug.py`:

```python
"""Demonstrate bug."""
import pylint.lint

pylint.lint.Run([__file__])

print("This line is missing from the output.")
```

### Command used

```shell
python bug.py
```

### Pylint output

```shell
$ python bug.py

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

```
...
This line is missing from the output.
```

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.9.10 (main, Jan 17 2022, 14:25:44) 
[GCC 7.5.0]
```

### OS / Environment

OpenSUSE Leap 15.2 Linux

**Comment by DanielNoord** [comment]
This is because `pylint` will exit with a `SystemExit` with this particular set of arguments.
You'll see that if you change the code to:
```python
"""Demonstrate bug."""
import pylint.lint

try:
    pylint.lint.Run([__file__])
except SystemExit:
    pass

print("This line is missing from the output.")
```

The line will get printed.

**Comment by bersbersbers** [comment]
I see, thanks! Related:
https://github.com/PyCQA/pylint/blob/2ab7d536159f5e54c195db7a42f00261c3b67827/pylint/lint/run.py#L370-L372

So this also works:

```python
"""Demonstrate workaround."""
import pylint.lint

pylint.lint.Run([__file__], exit=False)

print("This line is actually printed.")
```

I am now wondering if I should have seen that in the docs somewhere. If I should have, feel free to close this.

**Comment by DanielNoord** [comment]
Hmm, we don't really discuss running `pylint` from within a file in the documentation. I think we kind of assume that if you're doing that you'll likely look at the source code as well and might catch peculiarities likes this as well.

I'd be happy to review a PR if you got any good suggestions for a place in the documentation on where to discuss this, but if you're not up to that please close this. I'm not actually sure if it should be documented explicitly.

**Comment by bersbersbers** [comment]
Alright, agreed - thanks for the workarounds anyway!
